### ISSUE TITLE
Solidity 0.5.0 static binary is compiled as pre-release

### ISSUE THREAD

**Issue by silasdavis** [ethereum/solidity#5519]
Description:
The current solidity static binary (https://github.com/ethereum/solidity/releases/download/v0.5.0/solc-static-linux) reports itself as a pre-release version, see the output of solc --version:

```shell
$ solc --version
solc, the solidity compiler commandline interface
Version: 0.5.0+commit.1d4f565a.Linux.g++
```

The problem with this is solc treats any 'pre-release'/nightly as being less than any release version and will refuse to build files with pragma versions. So you get an error like:

```shell
 SyntaxError: Source file requires different compiler version 
(current compiler is 0.5.0+commit.1d4f565a.Linux.g++ 
- note that nightly builds are considered to be strictly less than the released version
```

This can happen if you build solidity from a commit that is ahead of the intended tag. However looking at the current tag it looks like this is referring to the correct commit: https://github.com/ethereum/solidity/releases/tag/v0.5.0 (commit hash: 1d4f565a).

Did your build system build in a context where the 0.5.0 tag was not applied?

**Comment by axic** [comment]
What is the version pragma you are using?

**Comment by silasdavis** [comment]
Sorry yeah you're right I jumped to conclusions - I didn't think the commit hash was included in the version of released versions, I was on `pragma solidity ^0.4.0;` `pragma solidity >=0.4.0;` is fine.

**Comment by VictorZhang2014** [comment]
We can specify an exact compiler version in `truffle.js` file, as doc mentioned https://www.trufflesuite.com/docs/truffle/reference/configuration#compiler-configuration

```
module.exports = {
  compilers: {
    solc: {
      version: "^0.6.0"
    }
  },

  networks: {
    development: {
      host: "127.0.0.1",
      port: 8545,
      network_id: "*" // Match any network id
    }
  }
};
```
This is my `truffle.js` configuration.
